### ISSUE TITLE
Specification OpenAPI - Field (exclusiveMinimum, exclusiveMaximum)

### ISSUE THREAD

**Issue by carlos-rian** [tiangolo/fastapi#3541]
## Spec Bug

I'm creating my documentation using "FastAPI". I'm having trouble importing the generated schema into Azure API Management. Because when I use the "gt" argument in the "Query" object, it defines the exclusiveMinimum field, in the OpenAPI documentation, this field is a boolean, but in FastAPI it is defined with Integer!

#### Spec OpenAPI:
* https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v3.0/schema.yaml
![image](https://user-images.githubusercontent.com/65134623/125998684-70be655f-b7dd-464b-85b0-9bba82847acb.png)
* https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#page-6
![image](https://user-images.githubusercontent.com/65134623/125997415-3b3c3e47-86b9-4e3e-b8e5-6dd51978af46.png)

#### Schema OpenAPI FastAPI
![image](https://user-images.githubusercontent.com/65134623/125997962-e0e79d74-a9e6-4201-bd48-d253b5d35488.png)

#### FastAPI Query
![image](https://user-images.githubusercontent.com/65134623/125998145-b7e93cee-ec94-45a6-8c8f-62802c0333f2.png)

#### FastAPI Docs
![image](https://user-images.githubusercontent.com/65134623/125998300-2e4f157e-3fb0-49af-88f6-a1798c161d55.png)

**Comment by JuanBenitezDev** [comment]
I found this is a Pydantic [issue](https://github.com/samuelcolvin/pydantic/issues/1164).

It seems it depends on the version of JSONSchema the OpenAPI is targeting.

**Comment by carlos-rian** [comment]
@JuanBenitezDev I understand, but I found that Spec OpenAPI V2 is also Boolean.

https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v2.0/schema.json
http://json-schema.org/draft-04/schema#

![image](https://user-images.githubusercontent.com/65134623/126008752-41435ada-39a4-43ab-8f01-85f58be6bc8e.png)

**Comment by JuanBenitezDev** [comment]
@carlos-rian Yeah I know, but Pydantic is generating schemas that are compliant with the last versions of the specifications. You can see that [here](https://pydantic-docs.helpmanual.io/usage/schema/)

The last version of OpenAPI at the time I'm writing this is [3.1](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md). This new version is based on the JSON Schema Draft 2020-12, here the exclusiveMinimum and exclusiveMaximum are numbers, you can see them [here](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.2.5)

![image](https://user-images.githubusercontent.com/22871775/126044736-75be5b71-90b0-49e3-aeb6-987aa95b299c.png)

You can even check what are the things changing from OpenAPI v3 to v3.1 [here](https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0)

My guess is that Azure API Management is expecting a specific OpenAPI version.

**Comment by Alexander-N** [comment]
See https://github.com/tiangolo/fastapi/issues/240 for more discussion. As workaround you could use `ge` instead of `gt`, e.g.
```python
Query(..., ge=11)
```
instead of
```python
Query(..., gt=10)
```

**Comment by carlos-rian** [comment]
@JuanBenitezDev I understand, it´s makes sense. Thanks you!

Hi @Alexander-N.  Yes. I've seen some discussions about this. I solved the problem using `ge`. But I thought it was a bug in Pydantic. Thanks!
